File: GTTS/INL.TrainingService.Models/ISaveTrainingEvent_Param.cs
using System;
using System.Collections.Generic;

namespace INL.TrainingService.Models
{
    public interface ISaveTrainingEvent_Param
    {
        Int64? TrainingEventID { get; set; }
        string Name { get; set; }
        string NameInLocalLang { get; set; }
        int TrainingEventTypeID { get; set; }
        string Justification { get; set; }
        string Objectives { get; set; }
        string ParticipantProfile { get; set; }
        string SpecialRequirements { get; set; }
        int? ProgramID { get; set; }
        int? KeyActivityID { get; set; }
        Int64 TrainingUnitID { get; set; }
        int? CountryID { get; set; }
        int? PostID { get; set; }
        int? ConsularDistrictID { get; set; }
        int? OrganizerAppUserID { get; set; }
        int? PlannedParticipantCnt { get; set; }
        int? PlannedMissionDirectHireCnt { get; set; }
        int? PlannedNonMissionDirectHireCnt { get; set; }
        int? PlannedMissionOutsourceCnt { get; set; }
        int? PlannedOtherCnt { get; set; }
        decimal? EstimatedBudget { get; set; }
        decimal? ActualBudget { get; set; }
        int? EstimatedStudents { get; set; }
        int? FundingSourceID { get; set; }
        int? AuthorizingLawID { get; set; }
        string Comments { get; set; }
        int? ModifiedByAppUserID { get; set; }

        List<SaveTrainingEventLocation_Item> TrainingEventLocations { get; set; }
        List<KeyActivity_Item> KeyActivities { get; set; }
        List<IAA_Item> IAAs { get; set; }
        List<SaveTrainingEventUSPartnerAgency_Item> TrainingEventUSPartnerAgencies { get; set; }
        List<SaveTrainingEventProjectCode_Item> TrainingEventProjectCodes { get; set; }
		List<SaveTrainingEventStakeholder_Item> TrainingEventStakeholders { get; set; }
	}
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.TrainingService.Models/isave-training-event_param.ts



import { SaveTrainingEventLocation_Item } from './save-training-event-location_item';
import { KeyActivity_Item } from './key-activity_item';
import { IAA_Item } from './iaa_item';
import { SaveTrainingEventUSPartnerAgency_Item } from './save-training-event-uspartner-agency_item';
import { SaveTrainingEventProjectCode_Item } from './save-training-event-project-code_item';
import { SaveTrainingEventStakeholder_Item } from './save-training-event-stakeholder_item';

export interface ISaveTrainingEvent_Param {
  
	TrainingEventID?: number;
	Name: string;
	NameInLocalLang: string;
	TrainingEventTypeID: number;
	Justification: string;
	Objectives: string;
	ParticipantProfile: string;
	SpecialRequirements: string;
	ProgramID?: number;
	KeyActivityID?: number;
	TrainingUnitID: number;
	CountryID?: number;
	PostID?: number;
	ConsularDistrictID?: number;
	OrganizerAppUserID?: number;
	PlannedParticipantCnt?: number;
	PlannedMissionDirectHireCnt?: number;
	PlannedNonMissionDirectHireCnt?: number;
	PlannedMissionOutsourceCnt?: number;
	PlannedOtherCnt?: number;
	EstimatedBudget?: number;
	ActualBudget?: number;
	EstimatedStudents?: number;
	FundingSourceID?: number;
	AuthorizingLawID?: number;
	Comments: string;
	TrainingEventLocations?: SaveTrainingEventLocation_Item[];
	KeyActivities?: KeyActivity_Item[];
	IAAs?: IAA_Item[];
	TrainingEventUSPartnerAgencies?: SaveTrainingEventUSPartnerAgency_Item[];
	TrainingEventProjectCodes?: SaveTrainingEventProjectCode_Item[];
	TrainingEventStakeholders?: SaveTrainingEventStakeholder_Item[];
}





